rails-commenting.rb

```diff
@@ -8,32 +8,40 @@
 
 # ---1)
 class BlogPostsController < ApplicationController
+# This is called a Ruby class which inherits from the ApplicationController and has methods.
+# BlogPostsController is calling from application Controller. In this example of a file we can see  BlogPostsController has many methods in this case the CRUD methods and is called upon the ApplicationController. Also as an analogy the controller is a middleman between models and views for the front end to retrieve and display while the back end can be used in the "back" of databases and applications.
   def index
     # ---2)
     @posts = BlogPost.all
+    #In this method it is used to display the collection of data inputted and made in the Ruby console. It is the source where all the data resides in i.e, a Database.  This is a GET requests from the RESTful CRUDs.
   end
 
   # ---3)
   def show
     @post = BlogPost.find(params[:id])
+    #In this method we are able retrieve specific data from the index database by using parameters of unique ids this is an Active Record call. In a very large database we can find particular data which makes searching by their IDs a lot easier rather than scrolling through information.
+    #This is a RESTful route.
   end
 
   # ---4)
   def new
     @post = BlogPost.new
+    #In this method we are able to add information on our web application and is stored in the database. This is part of RESTful as well and displays a form for users.
```

Comment:
Great job mentioning the form! Keep in mind the addition to the database will be in the create method.